api/utils/validations.js

```diff
@@ -0,0 +1,87 @@
+const { check, validationResult } = require('express-validator');
+
+exports.checkAddProduct =
+    [
+        check('title')
+            .trim()
+            .isLength({ min: 3 })
+            .withMessage('Minimum 3 characters required')
+            .bail(),
+        check('category')
+            .trim()
+            .isLength({ min: 3 })
+            .withMessage('Minimum 3 characters required')
+            .bail(),
+        check('color')
+            .trim()
+            .not()
+            .isEmpty()
+            .withMessage('Color can not be empty')
+            .bail(),
+        check('size')
+            .trim()
+            .not()
+            .isEmpty()
+            .withMessage('Size can not be empty')
+            .bail(),
+        check('price')
+            .trim()
+            .not()
+            .isEmpty()
+            .withMessage('Price can not be empty')
+            .isNumeric()
+            .withMessage('Price must be a number')
+            .bail(),
+        (req, res, next) => {
+            const errors = validationResult(req);
+            if (!errors.isEmpty())
+                return res.status(422).json({ errors: errors.array() });
```

Comment:
Also here you can create folder constants and there add status-codes.js. Add codes there for example:
```js
module.exports = {
    UnprocessableEntity: 422
    OK: 200,
    NotFound: 404
}
...

```
Avoid using magic values, whenever could export them just in case to reuse them. If changes need you will need to change it on one place.